### ISSUE TITLE
CheckBy causing Stack Overflow

### ISSUE THREAD

**Issue by Walach76** [wenzhixin/bootstrap-table#4495]
Hi, 

I just had a problem with the method CheckBy. It is quite easy to reproduce, for example, your data looks like that :
[{ Id : 1, Value : Toto }, {Id : 2, Value : Toto}]
When I check a row, I want to check all the rows with the same value.
But, by checking the second row in the current method, it rechecks the first row, and so on... which results in a stack overflow.
You may need to check if the row is already checked, and verify if there are elements to check.

On line 6525, I changed from this :
```javascript
if (obj.values.includes(row[obj.field])) {
    var $el = _this22.$selectItem.filter(':enabled').filter(Utils.sprintf('[data-index="%s"]', i)).prop('checked', checked);    
        row[_this22.header.stateField] = checked;
        rows.push(row);
        _this22.trigger(checked ? 'check' : 'uncheck', row, $el);
}
```

To this :
```javascript
if (obj.values.includes(row[obj.field])) {
    var $el = _this22.$selectItem.filter(':enabled').filter(Utils.sprintf('[data-index="%s"]', i)).not(':checked').prop('checked', checked);
    if ($el.length > 0) {
        row[_this22.header.stateField] = checked;
        rows.push(row);
        _this22.trigger(checked ? 'check' : 'uncheck', row, $el);
    }
}
```

I added .not(":checked") and if ($el.length > 0), which, in my case, solves the problem.

Note that by doing this, it will not fire the check event on already checked rows.

I think you can figure a better solution, for everyone.

Thanks !

**Comment by wenzhixin** [comment]
https://live.bootstrap-table.com/code/wenzhixin/377

**Comment by wenzhixin** [comment]
Fixed in #4528
